ui/core/_core-bs3.scss

```diff
@@ -108,6 +108,13 @@
       }
     }
   }
+
+  .slds-modal__header,
+  .slds-notify--toast {
+    h2 {
+      margin-top: $spacing-none;
+    }
+  }
```

Comment:
Im specifically targeting h2's because of the recommended SLDS markup for modals and notifications, but I dont think it would hurt anything.  Your call.